apis/python/src/tiledbsoma/soma_object.cc

```diff
@@ -63,7 +63,11 @@ void load_soma_object(py::module& m) {
                         uri, mode, context, timestamp, clib_type);
                 })();
 
-                auto soma_obj_type = soma_obj->type();
+                try {
+                    auto soma_obj_type = soma_obj->type();
```

Comment:
compile error since this isn't defined outside the try

also, IIUC this is a `std::optional<std:string>` so it would be nice to see explicit handling of the has-no-value case

Reply:
I get the same